lightning/src/chain/onchaintx.rs

```diff
@@ -404,10 +426,26 @@ impl<ChannelSigner: Sign> OnchainTxHandler<ChannelSigner> {
 			preprocessed_requests.push(req);
 		}
 
+		loop {
+			let pop_height = if let Some(first_entry) = self.locktimed_packages.iter().next() {
+				if *first_entry.0 <= height + 1 {
+					*first_entry.0
+				} else { break; }
+			} else { break; };
+			log_trace!(logger, "Restoring delayed claim of package(s) at their timelock at {}.", pop_height);
+			preprocessed_requests.append(&mut self.locktimed_packages.remove(&pop_height).unwrap());
+		}
+
 		// Generate claim transactions and track them to bump if necessary at
 		// height timer expiration (i.e in how many blocks we're going to take action).
 		for mut req in preprocessed_requests {
-			if let Some((new_timer, new_feerate, tx)) = self.generate_claim_tx(height, &req, &*fee_estimator, &*logger) {
+			if req.package_timelock() > height + 1 {
+				log_debug!(logger, "Delaying claim of package until its timelock at {} (current height {}), the following outpoints are spent:", req.package_timelock(), height);
+				for outpoint in req.outpoints() {
+					log_debug!(logger, "  Outpoint {}", outpoint);
+				}
+				self.locktimed_packages.entry(req.package_timelock()).or_insert(Vec::new()).push(req);
+			} else if let Some((new_timer, new_feerate, tx)) = self.generate_claim_tx(height, &req, &*fee_estimator, &*logger) {
```

Comment:
Can we broadcast delayed-transactions scenario with the following reorg-scenario ?
* delayed-transactions are cached in `locktimed_packages`
* height expiration is reached, we generate and broadcast delayed transactions
* chain is reorg-out, delayed transactions are evicted from mempools
* before syncing to new tip, we receive a preimage offchain and enter into `update_claims_view`

I don't think we can regenerate and broadcast a non-propagating delayed transaction as regeneration timer are height-based. In the future, if we switch regeneration timer to mempool feeration variations, we might broadcast such delayed transactions.

Otherwise, I can't think of other edgy reorg cases.

Reply:
In that case don't we still generate the relevant commitment + timeout/claim transactions when the block comes back in when we sync to the new tip? Either way we should bump after a few blocks if we haven't confirmed by then.

Reply:
Clearly, not a big deal. It's a just a edge-case about new feature added by this PR in case of a hypothetical future change. If we switch or add the option of mempool-based timing we'll have to split our hairs on few others things around OnchainTxHandler.

(Thought I still think mempool fluctuations-based timing might be more more interesting for fee savings than dumb rebroadcast after blocks X, if you're not near timelocks expiration)